state_machine/src/backend.rs

```diff
@@ -88,22 +82,18 @@ impl Backend for InMemory {
 		self.inner.update(changes);
 
 		// fully recalculate trie roots.
-
-		let storage_roots = self.inner.storage.iter().map(|(addr, storage)| {
+		use std::mem::transmute;
+		let storage_roots = self.inner.storage.iter().map(|(object, storage)| {
 			let flat_trie = storage.iter().map(|(k, v)| (k.to_vec(), v.clone())).collect();
-			(addr.to_vec(), sec_trie_root(flat_trie).to_vec())
+			(unsafe { transmute::<u64, [u8; 8]>(object.to_be()) }.to_vec(), sec_trie_root(flat_trie).to_vec())
```

Comment:
favor using the byteorder crate for these kinds of conversions

Reply:
yeah fair enough

Reply:
fixed. though now it's four lines plus a module import, whereas before it was actually perfectly safe and inline.